packages/next-on-pages/src/buildApplication/processVercelFunctions/dedupeEdgeFunctions.ts

```diff
@@ -495,6 +497,45 @@ function fixFunctionContents(contents: string): string {
 	return contents;
 }
 
+/**
+ * Server actions check the request headers.origin url host and try to make sure that that is
+ * the same as that of other Next.js-specific headers (this is a measure against CSRF attacks)
+ *
+ * This resulted problematic when trying to run the application with `wrangler pages dev` as the
+ * two hosts can get misaligned (one being `localhost:<port>` while the other `127.0.0.1:<port>`)
```

Comment:
I kind of feel like this is a problem with `wrangler pages dev` no?
Why is there this misalignment? Could you expand on that?

Reply:
That's a totally valid point, I am not sure if anything can be changes in wrangler but I can check with the team

The problem is that this issue is caused by a fundamental design aspect of the `startDevWorker` changes

Basically `wrangler pages dev` gives you the proxy worker's address (e.g. `http://localhost:8788`) but the actual worker is running on a different port (and not on `localhost`) (e.g. `http://127.0.0.1:51405/`) so, when, with the browser you to go the address `wrangler pages dev` provides you and try to make a request, the host from the browser (which is the proxy worker's address) is different from the actual worker's address (which is not really visible by the user) and that's what causing the misalignment.

So if in the browser I were to go to the actual worker's address (e.g. `http://127.0.0.1:51405/server-actions/simple-kv-forms`) things would just work as expected, the issues there are that:
 - the address is not provided to you by wrangler (the way I find it is by going to the standard proxy address, trying to use action and checking the terminal error and picking up the address from that)
 - the address is not stable and (as far as I know) changes whenever the worker code changes